### PR TITLE
Refactoring and support cyrillic language wrap

### PR DESCRIPTION
**Pull request by zilongshanren**
https://github.com/cocos-creator/fireball/issues/3909

Changes proposed in this pull request:
- Refactoring, abstract a TextUtils for common text operations
- support cyrillic text wrap

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/label/CCTextUtils.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    isCJK_unicode: function(ch) {
+        var __CHINESE_REG = /^[\u4E00-\u9FFF\u3400-\u4DFF]+$/;
+        var __JAPANESE_REG = /[\u3000-\u303F]|[\u3040-\u309F]|[\u30A0-\u30FF]|[\uFF00-\uFFEF]|[\u4E00-\u9FAF]|[\u2605-\u2606]|[\u2190-\u2195]|\u203B/g;
+        var __KOREAN_REG = /^[\u1100-\u11FF]|[\u3130-\u318F]|[\uA960-\uA97F]|[\uAC00-\uD7AF]|[\uD7B0-\uD7FF]+$/;
+        return __CHINESE_REG.test(ch) || __JAPANESE_REG.test(ch) || __KOREAN_REG.test(ch);
+    },
+
+    //Checking whether the character is a whitespace
+    isspace_unicode: function(ch) {
```
Comment (jareguo) [inline cocos2d/core/label/CCTextUtils.js]: 为什么 space 不是首字母大写

**Review by jareguo — APPROVED** [review]

File: cocos2d/core/label/CCTextUtils.js (hunk 1 above)
Comment (zilongshanren) [reply cocos2d/core/label/CCTextUtils.js]: 我改成 isUnicodeSpace 吧？

File: cocos2d/core/label/CCTextUtils.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+var TextUtils = {
+    label_wordRex : /([a-zA-Z0-9ÄÖÜäöüßéèçàùêâîôûа-яА-ЯЁё]+|\S)/,
+    label_symbolRex : /^[!,.:;}\]%\?>、‘“》？。，！]/,
+    label_lastWordRex : /([a-zA-Z0-9ÄÖÜäöüßéèçàùêâîôûа-яА-ЯЁё]+|\S)$/,
+    label_lastEnglish : /[a-zA-Z0-9ÄÖÜäöüßéèçàùêâîôûа-яА-ЯЁё]+$/,
+    label_firsrEnglish : /^[a-zA-Z0-9ÄÖÜäöüßéèçàùêâîôûа-яА-ЯЁё]/,
+    label_wrapinspection : true,
+
+    isCJK_unicode: function(ch) {
+        var __CHINESE_REG = /^[\u4E00-\u9FFF\u3400-\u4DFF]+$/;
```
Comment (zilongshanren) [inline cocos2d/core/label/CCTextUtils.js:35]: 这个函数也改成  isUnicodeCJK?

Comment (jareguo) [reply cocos2d/core/label/CCTextUtils.js:35]: 同意

File: cocos2d/core/label/CCTextUtils.js (hunk 1 above)
Comment (jareguo) [reply cocos2d/core/label/CCTextUtils.js]: 同意

**Comment by zilongshanren** [comment]
@jareguo  done

**Pull request merged**